source/plot/LinePlotData2D.ooc

```diff
@@ -27,14 +40,35 @@ LinePlotData2D: class extends PlotData2D {
 			result = result & "<path stroke='" clone() & this color clone() & "' stroke-opacity='" clone() & this opacity toString() & "' fill='none' stroke-width='" clone() & this lineWidth toString() & "' d='M " clone() & (scaling x * this dataSeries[0] x) toString() & " " clone() & (- scaling y * this dataSeries[0] y) toString() & " L " clone()
 			for (j in 1..this dataSeries count)
 				result = result & (scaling x * this dataSeries[j] x) toString() & " " clone() & (- scaling y * this dataSeries[j] y) toString() & " " clone()
-			result = result & "'/>\n" clone()
+			result = result >> "' "
+			match (this lineStyle) {
+				case LineStyle Dashed =>
+					result = result >> "stroke-dasharray='" & (this lineWidth * 5) toString() >> "," & (this lineWidth * 5) toString() >> "'"
+				case LineStyle Dotted =>
+					result = result >> "stroke-dasharray='" & this lineWidth toString() >> "," & this lineWidth toString() >> "'"
+				case =>
+					// do not do anything, will result in solid line
+			}
+			result = result >> "/>\n"
 		}
 		result
 	}
 
-	getSvgLegend: func (legendCount: Int) -> String {
-		boundaryOffset := 5
-		result := "<text id='" clone() & this label clone() & "' x='" clone() & boundaryOffset toString() & "' y='" clone() & (this fontSize * legendCount + boundaryOffset) toString() & "' font-size='" clone() & this fontSize toString() & "' fill='" clone() & this color clone() & "' fill-opacity='" clone() & this opacity toString() & "'> ▬ " clone() & this label clone() & "</text>\n" clone()
+	getSvgLegend: func (legendCount, fontSize: Int) -> String {
+		result: String
+		start := FloatPoint2D new(this legendOffset, this legendOffset + (fontSize * legendCount - fontSize / 2) as Float)
+		end := FloatPoint2D new(this legendOffset + fontSize, start y)
+		match (this lineStyle) {
+			case LineStyle Solid =>
+				result = Shapes line(start, end, this lineWidth, this opacity, this color)
```

Comment:
Same here, this case is the same as the default case. Leave a comment in the default case `case => // Solid` if you want.